jina/helper.py

```diff
@@ -503,6 +503,11 @@ def valid_yaml_path(path: str, to_stream: bool = False):
                                 ' or a valid file path, or a supported class name.' % path)
 
 
+def valid_docker_image(image: str):
+    # TODO: Think a better way, it is a little dangerous
+    return image.startswith('jinaai/') or image.startswith('jina/')
+
```

Comment:
```
def valid_docker_image(repo_name, version="latest"):
    try:
        import docker
        import requests
    except:
        return False
    client = docker.from_env()
    try:
        client.images.get(repo_name)
        return True
    except:
        pass
    url = "https://index.docker.io/v1/repositories/" + repo_name + "/tags/" + version
    try:
        res = requests.get(url)
        return res.status_code != 404
    except:
        return False
```

Reply:
hey @YueLiu-jina thank you very much it seems like an interesting approach, can you refactor in a single try ... except catching different kinds of exceptions?

Reply:
the only problem is that the signature does not match how docker image name will be inserted to the Flow. It would come as a single string.

Reply:
code edited @JoanFM

Reply:
good change, I suggest, remove the recursion and just reassign version and repo_name and go hit the try-catch. recursion is  sometimes a little dangerous.

Reply:
@YueLiu-jina , maybe u can do a Pull Request to my branch with your proposed ideas, I think they are good.